NewPlatform.Flexberry.ServiceBus.RabbitMQTests/DefaultRmqComponentsTest.cs

```diff
@@ -0,0 +1,293 @@
+                var collector = new RmqStatisticsCollector(
+                    GetMockLogger(),
+                    esbSubscriptionsManager,
+                    mockStatisticsSetting.Object,
+                    managementClient,
+                    amqpNamingManager,
+                    mockStatisticsSaveService.Object,
+                    testVhost);
+                collector.Interval = StatisticsInterval.TenSeconds;
+
+                collector.Prepare();
+                collector.Start();
+                Thread.Sleep(11000);
+
+                Assert.Contains(info, x => x.SentCount == 2
+                    && x.ReceivedCount == 2
+                    && x.QueueLength == 0
+                    && x.StatisticsSetting.Subscription.Client.ID == WithoutCallbackReceiverId
+                    && x.StatisticsSetting.Subscription.MessageType.ID == WithoutCallbackMsgTypeId);
+
+                Assert.Contains(info, x => x.SentCount == 2
+                    && x.ReceivedCount == 1
+                    && x.QueueLength == 0
+                    && x.StatisticsSetting.Subscription.Client.ID == CallbackReceiverId
+                    && x.StatisticsSetting.Subscription.MessageType.ID == CallbackMsgTypeId);
+
+                rmqReceivingManager.AcceptMessage(callbackSbMessage1);
+                rmqReceivingManager.AcceptMessage(callbackSbMessage2);
+                rmqReceivingManager.AcceptMessage(withoutCallbackSbMessage1);
+                rmqReceivingManager.AcceptMessage(withoutCallbackSbMessage2);
+
+                var message2 = rmqSendingManager.ReadMessage(WithoutCallbackReceiverId, WithoutCallbackMsgTypeId);
+                rmqSendingManager.DeleteMessage(message.__PrimaryKey.ToString());
+                Thread.Sleep(11000);
+
+                Assert.Contains(info, x => x.SentCount == 2
+                    && x.ReceivedCount == 2
+                    && x.QueueLength == 0
+                    && x.StatisticsSetting.Subscription.Client.ID == WithoutCallbackReceiverId
+                    && x.StatisticsSetting.Subscription.MessageType.ID == WithoutCallbackMsgTypeId);
+
+                Assert.Contains(info, x => x.SentCount == 2
+                    && x.ReceivedCount == 1
+                    && x.QueueLength == 0
+                    && x.StatisticsSetting.Subscription.Client.ID == CallbackReceiverId
+                    && x.StatisticsSetting.Subscription.MessageType.ID == CallbackMsgTypeId);
+            }
+        }
+
+        private void ClearStatistics()
+        {
+            string userName = ConfigurationManager.AppSettings["DefaultRmqUserName"];
+            string password = ConfigurationManager.AppSettings["DefaultRmqUserPassword"];
+            string host = ConfigurationManager.AppSettings["DefaultRmqHost"];
+            string managementPort = ConfigurationManager.AppSettings["DefaultRmqManagementPort"];
+
+            string sURL = $"http://{host}:{managementPort}/api/reset";
```

Comment:
There is no such method in ManagementClient?

Reply:
As far as i know, the only way you can reset RabbitMQ statistics is by using either rabbitmqctl or HTTP API.